aospy/calc.py

```diff
@@ -334,123 +371,123 @@ def _get_all_data(self, start_date, end_date):
+                                                  self.var.description,
+                                                  self.dtype_out_vert)
+                reduced.update({reduc.label: data_out})
         return OrderedDict(sorted(reduced.items(), key=lambda t: t[0]))
 
-    def compute(self, write_to_tar=True):
-        """Perform all desired calculations on the data and save externally."""
-        data = self._get_all_data(self.start_date, self.end_date)
-        logging.info('Computing timeseries for {0} -- '
-                     '{1}.'.format(self.start_date, self.end_date))
-        full, full_dt = self._compute_full_ts(data)
-        full_out = self._full_to_yearly_ts(full, full_dt)
-        reduced = self._apply_all_time_reductions(full_out)
-        logging.info("Writing desired gridded outputs to disk.")
-        for dtype_time, data in reduced.items():
-            data = _add_metadata_as_attrs(data, self.var.units,
-                                          self.var.description,
-                                          self.dtype_out_vert)
-            self.save(data, dtype_time, dtype_out_vert=self.dtype_out_vert,
-                      save_files=True, write_to_tar=write_to_tar)
+    # @log_step(logging.info, ('Executing calculation for the dates {0} -- '
+    #                          '{1}.'.format(self.start_date, self.end_date)))
+    def compute(self, write_to_netcdf=True, write_to_tar=True):
+        """Perform all desired calculations on the data and save externally.
+
+        This method goes through each step needed to perform the given
+        calculations:
+
+        1. Load all required data from disk.
+        2. Subset the data to the desired times.
+        3. Compute the desired quantity.
+        4. Apply all desired spatiotemporal reductions.
+        5. Save the results to this object's ``data_out`` attribute.
+        6. If specified, write the results to disk.
+
+        Parameters
+        ----------
+        write_to_netcdf: bool, optional, write_to_tar : bool, optional
+            Whether to write the results of the calculation to netCDF files on
+            disk.  Default True.
+        write_to_tar: bool, optional
+            Whether to write the results of the calculation to a tarball
+            archive.  Default True.
+
+        Returns
+        -------
+        This ``Calc`` object, whose ``data_out`` attribute has been created or
+        updated with the results of the calculations performed.
+
+        """
+        # Load.
+        data_in = self._get_all_data(self.start_date, self.end_date)
+        # Compute.
+        data_out = self._function(*data_in).rename(self.name)
+        data_out = _time_weights_as_days(data_out)
+        # Reduce.
+        data_out = self._apply_vert_reduc(data_out)
+        gridded_out = self._apply_gridded_reductions(data_out)
+        region_out = self._apply_region_reductions(data_out)
+        combined_out = {**gridded_out, **region_out}
```

Comment:
> Hmmm not sure what's up with the stickler comments...all tests are passing on my machine.

I think this syntax requires Python 3.5 or above (this is fine since we have dropped Python 2 support).  Maybe modifying our `.stickler.yml` to look like the following could do the trick?
```
linters:
  flake8:
    python: 3
    fixer: true
```
See here for more information: https://stickler-ci.com/docs#python.  `python: 3` means Python 3.6 for Stickler.

Reply:
Good catch!  That did the trick.